status-page.py

```diff
@@ -92,8 +101,8 @@ def update_incident_status(self):
             version_and_name = f'{api_version}/{api_name}'
```

Comment:
I think some logic is repeated between `passed_tests` and `failed_tests`. Consider refactoring codes to make them reuseable.